platform/net/zephyr/src/mender-http.c

```diff
@@ -304,6 +304,11 @@ mender_http_artifact_download(const char *uri, mender_artifact_download_data_t *
         goto END;
     }
 
+    /* Artifact download failed */
+    if (MENDER_OK != dl_data->ret) {
+        goto END;
+    }
```

Comment:
Please check if we need to change this in the curl-based implementation as well.